newIDE/app/src/EventsSheet/EventsTree/DropContainer.js

```diff
@@ -0,0 +1,300 @@
+      }}
+    >
+      <DropTargetContainer
+        style={{
+          dropIndicator: indicatorStyles['top'],
+          dropArea: dropAreaStyles['top'],
+        }}
+        onDrop={() => onDrop(moveNodeAbove, node)}
+        {...commonProps}
+      />
+      {canHaveSubEvents && canDropBelow && (
+        <>
+          <DropTargetContainer
+            style={{
+              dropIndicator: indicatorStyles['bottom-right'],
+              dropArea: dropAreaStyles['bottom-right'],
+            }}
+            onDrop={() => onDrop(moveNodeAsSubEvent, node)}
+            {...commonProps}
+          />
+          <DropTargetContainer
+            style={{
+              dropIndicator: indicatorStyles['bottom'],
+              dropArea: dropAreaStyles['bottom-left'],
+            }}
+            onDrop={() => onDrop(moveNodeBelow, node)}
+            {...commonProps}
+          />
+          {/* Allow dragging left/right to move below or as subevent */}
+          {(isDraggedNodesOnlyChild || isDraggedNodeSameDepthAndBelow) && (
+            <>
+              <DropTargetContainer
+                style={{
+                  dropIndicator: indicatorStyles['bottom'],
+                  dropArea: dropAreaStyles['bottom-bottom-left'],
+                }}
+                onDrop={() => onDrop(moveNodeBelow, node)}
+                {...commonProps}
+              />
+              <DropTargetContainer
+                style={{
+                  dropIndicator: indicatorStyles['bottom-right'],
+                  dropArea: dropAreaStyles['bottom-bottom-right'],
+                }}
+                onDrop={() => onDrop(moveNodeAsSubEvent, node)}
+                {...commonProps}
+              />
+            </>
+          )}
+        </>
+      )}
+      {!canHaveSubEvents && canDropBelow && (
+        <DropTargetContainer
+          style={{
+            dropIndicator: indicatorStyles['bottom'],
+            dropArea: dropAreaStyles['bottom'],
+          }}
+          onDrop={() => onDrop(moveNodeBelow, node)}
+          {...commonProps}
+        />
+      )}
+      {canHaveSubEvents && !canDropBelow && (
+        <DropTargetContainer
+          style={{
+            dropIndicator: indicatorStyles['bottom-right'],
+            dropArea: dropAreaStyles['bottom'],
+          }}
+          onDrop={() => onDrop(moveNodeAsSubEvent, node)}
+          {...commonProps}
+        />
+      )}
+    </div>
+  );
+}
+
+export function AutoScroll({
+  direction,
+  DnDComponent,
+  activateTargets,
+  onHover,
+}: {|
+  direction: 'top' | 'bottom',
+  DnDComponent: DropTargetComponent<SortableTreeNode>,
+  activateTargets: boolean,
+  onHover: () => void,
+|}) {
+  const delayActivationTimer = React.useRef<?TimeoutID>(null);
+  const [show, setShow] = React.useState(false);
+
+  // This drop target overlaps with sibling drag source and cancels drag immediately.
+  // See: https://github.com/react-dnd/react-dnd/issues/766#issuecomment-388943403
```

Comment:
Very nice to leave these comments to github issues, will be useful should this break in the future.